File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s):
    stk = []
    for c in s:
        if not stk:
            stk.append(c)
        else:
            if stk[-1] == c:
                stk.pop()
            else:
                stk.append(c)
    if stk:
        return 0
    else:
        return 1

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.cpp
#include <iostream>
#include<string>
#include <vector>

using namespace std;

int solution(string s)
{
    int answer = -1;
    vector<char> stk;
    for(char c:s){
        if (!stk.empty()){
            if(stk.back()==c){
                stk.pop_back();
            }
            else{
                stk.push_back(c);
            }
        }
        else{
            stk.push_back(c);
        }
    }
    if(stk.empty()) return 1;
    return 0;
    return answer;
}